app/views/locations/index.html.erb

```diff
@@ -55,18 +55,22 @@
           data: {
             latitude: @search.result[:latitude],
             longitude: @search.result[:longitude],
-            'marker-icon-path': asset_path('locator/marker-pppo.svg'),
-            'marker-icon-2x-path': asset_path('locator/marker-pppo.svg'),
+            'marker-transportation-office-icon-path': asset_path('locator/marker-pppo.svg'),
+            'marker-transportation-office-icon-2x-path': asset_path('locator/marker-pppo.svg'),
+            'marker-weight-scale-icon-path': asset_path('locator/marker-weight-scale.svg'),
+            'marker-weight-scale-icon-2x-path': asset_path('locator/marker-weight-scale.svg'),
             'marker-shadow-path': asset_path('leaflet/dist/images/marker-shadow.png')
           }
     %>
 
-    <p>Displaying results near <b><%= @search.query %></b><%= " (page #{@transportation_offices.current_page} of #{@transportation_offices.total_pages})" if @transportation_offices.total_pages.positive? %>.</p>
+    <p>Displaying results near <b><%= @search.query %></b><%= " (page #{@locations.current_page} of #{@locations.total_pages})" if @locations.total_pages.positive? %>.</p>
 
     <div id="search-results">
-      <%= render partial: 'transportation_office', collection: @transportation_offices %>
+      <%- @locations.each do |location| -%>
+        <%= render partial: location.locatable_type.underscore, object: location.locatable %>
```

Comment:
Dynamically choosing the partial here based on the "locatable" class name…

- `TransportationOffice` => `transportation_office`
- `WeightScale` => `weight_scale`